src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -98,36 +106,95 @@ p5.RendererGL.prototype.vertex = function(){
  */
 p5.RendererGL.prototype.endShape =
 function(mode, isCurve, isBezier,isQuadratic, isContour, shapeKind){
-
-  var gl = this.GL;
-  var shader = this.curShader;
-  if (shader === this._getColorShader()) {
-    // this is the fill/stroke shader for retain mode.
+  if(this.drawMode !== constants.TEXTURE) {
     // must switch to immediate mode shader before drawing!
-    shader = this.setShader(this._getImmediateModeShader());
-
+    this.setFillShader(this._getImmediateModeShader());
     // note that if we're using the texture shader...
     // this shouldn't change. :)
   }
-  shader.bindShader();
+
+  if(this.curStrokeShader.active === true) {
+    for(var i=0; i<this.immediateMode.vertexPositions.length; i++) {
+      if(i+1 < this.immediateMode.vertexPositions.length) {
```

Comment:
`if (i < this.immediateMode.vertexPositions.length - 1) {` is a little more conventional way to write this